elastic/src/main/java/org/openmbee/mms/elastic/CommitElasticDAOImpl.java

```diff
@@ -35,27 +36,70 @@ public void indexAll(Collection<? extends BaseJson> jsons) {
-        this.index(getIndex(), json);
+        index((CommitJson) json);
+    }
+
+    public void index(CommitJson json) {
+        int commitCount = getCommitSize(json);
+        List<CommitJson> broken = new ArrayList<>();
+        if (commitCount > indexLimit) {
+            List<Map<String, Object>> allActions = new ArrayList<>();
+            allActions.addAll(json.getAdded().stream().peek(toAdd -> toAdd.put("action", "added")).collect(Collectors.toList()));
+            allActions.addAll(json.getUpdated().stream().peek(toUpdate -> toUpdate.put("action", "updated")).collect(Collectors.toList()));
+            allActions.addAll(json.getDeleted().stream().peek(toDelete -> toDelete.put("action", "deleted")).collect(Collectors.toList()));
+
+            while (!allActions.isEmpty()) {
+                CommitJson currentCommitCopy = CommitJson.copy(json);
+                while (getCommitSize(currentCommitCopy) < indexLimit && !allActions.isEmpty()) {
+                    Map<String, Object> action = allActions.remove(0);
+                    switch (action.getOrDefault("action", "none").toString()) {
+                        case "added":
+                            currentCommitCopy.getAdded().add(action);
+                            break;
+                        case "updated":
+                            currentCommitCopy.getUpdated().add(action);
+                            break;
+                        case "deleted":
+                            currentCommitCopy.getDeleted().add(action);
+                            break;
+                    }
+                }
+                broken.add(currentCommitCopy);
+
+            }
+            this.indexAll(broken);
+
+        } else {
+            this.index(getIndex(), json);
+        }
     }
 
-    public Optional<CommitJson> findById(String docId) {
-        return this.findById(getIndex(), docId);
+    public Optional<CommitJson> findById(String commitId) {
+        return Optional.of(getFullCommit(commitId));
     }
 
-    public List<CommitJson> findAllById(Set<String> docIds) {
-        return this.findAllById(getIndex(), docIds);
+    public List<CommitJson> findAllById(Set<String> commitIds) {
+        return getFullCommits(commitIds);
     }
 
-    public void deleteById(String docId) {
-        this.deleteById(getIndex(), docId);
+    public void deleteById(String commitId) {
+        List<CommitJson> docs = getDocs(commitId);
+        docs.forEach(commit -> {
+           this.deleteById(getIndex(), commit.getDocId());
+        });
     }
 
     public void deleteAll(Collection<? extends BaseJson> jsons) {
         this.deleteAll(getIndex(), jsons);
     }
 
-    public boolean existsById(String docId) {
-        return this.existsById(getIndex(), docId);
+    public boolean existsById(String commitId) {
+        List<CommitJson> docs = getDocs(commitId);
+        List<Boolean> result = new ArrayList<>();
+        docs.forEach(commit -> {
```

Comment:
isn't it enough to check the list isn't empty? the fact it returns the object means it exists?